pybricks/common/pb_type_speaker.c

```diff
@@ -22,34 +23,90 @@
 #include <pybricks/util_mp/pb_obj_helper.h>
 #include <pybricks/util_pb/pb_error.h>
 
+#define MP_PI MICROPY_FLOAT_CONST(3.14159265358979323846)
+
 typedef struct {
     mp_obj_base_t base;
     bool intialized;
 } pb_type_Speaker_obj_t;
 
 STATIC pb_type_Speaker_obj_t pb_type_Speaker_singleton;
 
+STATIC uint8_t volume = 100; // volume in 0..100 range
+
+// The number to multiply the sample amplitude by, to amplify the signal by the defined volume.
+// The original sample amplitude must be in the -1..1 range.
+STATIC uint16_t sample_amplifier = INT16_MAX;
+
 STATIC uint16_t waveform_data[128];
 
+// The amount of radians each PCM sample represents to completely fill whe waveform buffer for 2*PI
+//STATIC mp_float_t pcm_sample_size_rad = MP_PI * 2 / MP_ARRAY_SIZE(waveform_data);
+
+STATIC mp_obj_t pb_type_Speaker_get_volume(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    return mp_obj_new_int(volume);
```

Comment:
Micro-optimization (works since we know volume can never exceed `MP_SMALL_INT_MAX`
```suggestion
    return MP_OBJ_NEW_SMALL_INT(volume);
```

Reply:
Done.